sdk/include/platform/ibex/platform-dma.hh

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#define MALLOC_QUOTA 0x100000
+#define TEST_NAME "Allocator"
+#include "tests.hh"
+
+#include <cheri.hh>
+#include <compartment-macros.h>
+#include <cstdint>
+#include <stdint.h>
+#include <utils.hh>
+
+#include <thread.h>
+#include <thread_pool.h>
+
+using thread_pool::async;
+DECLARE_AND_DEFINE_ALLOCATOR_CAPABILITY(dmaDriverHeap, 8144);
+
+namespace Ibex {
+    class GenericDMA {
+        private:
+
+        Timeout noWait{0};
+        SObjStruct *dmaHeap;
+
+        struct StreamingInterface
+        {
+            uint32_t control;
+            uint32_t status;
+            uint32_t address;
+            uint32_t lengthInBytes;
+        };
+
+        struct DMAInterface
+        {
+            /**
+             *  these two streaming interfaces are for 
+             *  [0] memory to target and 
+             *  [1] target to memory directions.
+             *  target here is any I/O device, peripheral or an accelerator
+             */
+            volatile StreamingInterface *streamingInterface[2];
+        };
+
+        volatile DMAInterface *dmaInterface;
```

Comment:
Caching this will be slower than using the `MMIO_CAPABILITY` directly.  If you want to save some typing, wrapping the `MMIO_CAPABILITY` use in an `__always_inline` function will be better.